nemo/utils/exp_manager.py

```diff
@@ -536,22 +547,14 @@ def configure_checkpointing(trainer: 'pytorch_lightning.Trainer', log_dir: Path,
     else:
         logging.warning("trainer had a weights_save_path of cwd(). This was ignored.")
     # Create the callback and attach it to trainer
-    class NeMoModelCheckpoint(ModelCheckpoint):
-        @rank_zero_only
-        def on_train_end(self, trainer, pl_module):
-            filepath = os.path.join(self.dirpath, self.prefix + 'end.ckpt')
-            # TODO: Remove try, except block once lightning's ModelCheckpoint is stable
-            try:  # Try lightning master signature
-                self._save_model(filepath, trainer, pl_module)  # noqa pylint: disable=too-many-function-args
-            except TypeError:  # Fall back to lightning == 0.8.5 signature if failed
-                self._save_model(filepath)  # noqa
 
     checkpoint_callback = NeMoModelCheckpoint(
         filepath=Path(log_dir / 'checkpoints' / '{val_loss:.2f}-{epoch}'),
         save_top_k=3,
+        monitor='val_loss',
```

Comment:
Need to find a way to pass these parameters to the user through the yaml

Reply:
"val_loss" is not a given - RNNT can optionally disable it cause it is non-trivial cost to compute it over entire datasets (expensive Prediction step and Joint step cost). The default should be loss because that is understood to always be available (since backprop cannot occur without it).

Reply:
Back in 0.9.0, lightning used to assume that "val_loss" was the key that people returned from their validation step.

While moving in a more general director makes sense for lightning, should we do so for NeMo? Or should we just enforce model users to good defaults? If they choose not to use the default, then they have to manually adjust their scripts to make use of things like the ModelCheckpoint callback.

Reply:
So currently ModelPT has no access to ExpManager (it's called before ModelPT) nor can it ensure the list of callbacks is in the correct order to override it at a later point. Even if it could parse the callback list, it can't access any of the path values as exp manager is a stateless function. 

For now, it's fine to keep it val_loss, and provide an exp manager override to chose the monitor. I can override that in RNNT configs.

Reply:
Also, good defaults is very model dependent. An experiment manager should offer the flexibility without having to completely override one of it's core tasks

Reply:
Yes I think the correct way is to leave "val_loss" as default and provide an override via it's yaml config. I'll think on it more once we merge this PR